### PR TITLE
downsampling, removing padding, adding initial scoring script

### PR DESCRIPTION
**Pull request by aarora8**
adding modifications for downsampling image. It will help in runtime speedup, saving ram and disk space.

### CODE REVIEW

File: egs/madcat_arabic/v1/path.sh [hunk 1]
```diff
@@ -1,5 +1,7 @@
-export PATH="${HOME}/miniconda3/bin:$PATH"
-export PYTHONPATH="${PYTHONPATH}:${HOME}/miniconda3"
+#!/usr/bin/env bash
+
+export PATH="${HOME}/anaconda3/bin:$PATH"
```
Comment (danpovey) [inline egs/madcat_arabic/v1/path.sh]: did you intend to commit this change?

Comment (aarora8) [reply egs/madcat_arabic/v1/path.sh]: sorry, no. i will revert it.

File: egs/madcat_arabic/v1/local/generate_mask_from_page_image.py [hunk 2]
```diff
@@ -76,32 +81,41 @@ def update_minimum_bounding_box_input(bounding_box_input):
...
         new_x = x + offset
         new_y = y + offset
-        word_coordinate = (new_x, new_y)
-        updated_minimum_bounding_box_input.append(word_coordinate)
+        new_point = (new_x, new_y)
+        paded_mbb.append(new_point)
+
+    resized_mbb = []
+    ratio = int(args.downsampling_ratio)
+    for point in paded_mbb:
```
[the remark this answers is not in the document]
Comment (aarora8) [reply egs/madcat_arabic/v1/local/generate_mask_from_page_image.py]: ok thanks, will do

File: egs/madcat_arabic/v1/local/generate_mask_from_page_image.py [hunk 3]
```diff
@@ -76,32 +81,41 @@ def update_minimum_bounding_box_input(bounding_box_input):
...
+    image_file_name (string): complete path and name of the page image.
+    madcat_file_path (string): complete path and name of the madcat xml file
+                                  corresponding to the page image.
+    """
     im_wo_pad = Image.open(image_file_name)
-    im = pad_image(im_wo_pad)
-    im_arr = np.array(im)
+    im_pad = pad_image(im_wo_pad)
+    im_resized = downsample_image(im_pad)
+    im_arr = np.array(im_resized)
```
[the remark this answers is not in the document]
Comment (aarora8) [reply egs/madcat_arabic/v1/local/generate_mask_from_page_image.py]: I was downsizing the objects in update_minimum_bounding_box_input method. I will use the interface now.

File: egs/madcat_arabic/v1/local/generate_mask_from_page_image.py [hunk 4]
```diff
@@ -53,6 +56,8 @@
                     help='directory location to write output files')
 parser.add_argument('--padding', type=int, default=400,
                     help='padding across horizontal/verticle direction')
+parser.add_argument('--downsampling_ratio', type=int, default=2,
```
[the remark this answers is not in the document]
Comment (aarora8) [reply egs/madcat_arabic/v1/local/generate_mask_from_page_image.py]: ok thanks. approximately what size should be considered for max-image-size.

File: egs/madcat_arabic/v1/scoring/scoring_utils.py [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+        gt_pol_points = []
+        det_pol_points = []
+
+        pairs = []
+        det_matched_nums = []
+
+        pointlist = _get_pointlist(gt_file)
+        for n in range(len(pointlist)):
+            points = pointlist[n]
+            gtRect = Rectangle(*points)
```
Comment (danpovey) [inline egs/madcat_arabic/v1/scoring/scoring_utils.py]: be careful about naming conventions.  Shouldn't this be gt_rect?  And I don't like 'gt' as part of a name-- try to use full  words.

File: egs/madcat_arabic/v1/scoring/scoring_utils.py [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+            gt_pol = _rectangle_to_polygon(gtRect)
+            gt_pol = _polygon_from_points(points)
+            gt_pols.append(gt_pol)
+            gt_pol_points.append(points)
+
+        if result_file in subm:
```
Comment (danpovey) [inline egs/madcat_arabic/v1/scoring/scoring_utils.py]: this code is kind of unstructured and undocumented.  please improve it when you have time.

**Pull request merged**